src/tracker/inflights.rs

```diff
@@ -37,13 +42,41 @@ impl Inflights {
             start: 0,
             count: 0,
             cap,
+            incoming_cap: None,
+        }
+    }
+
+    /// Adjust inflight buffer capacity. Set it to `0` will disable the progress.
+    // Calling it between `self.full()` and `self.add()` can cause a panic.
+    pub fn set_cap(&mut self, incoming_cap: usize) {
+        match self.cap.cmp(&incoming_cap) {
+            Ordering::Equal => self.incoming_cap = None,
+            Ordering::Less => {
+                self.cap = incoming_cap;
```

Comment:
We use a ring buffer mechanism, and calculate the next index base on `self.cap` in `fn add`, direct changing `self.cap` may cause us to get a wrong next index.